gossipd/routing.c

```diff
@@ -566,8 +611,12 @@ const struct short_channel_id *handle_channel_announcement(
 					   &pending->node_id_2,
 					   &pending->bitcoin_key_1,
 					   &pending->bitcoin_key_2)) {
+		err = towire_errorfmt(rstate, NULL,
+				      "Malformed channel_announcement %s",
+				      tal_hex(pending, pending->announce));
 		tal_free(pending);
-		return NULL;
+		*scid = NULL;
```

Comment:
Having these null-sets plastered all over the place seems dangerous, wouldn't it not be better to just set NULL once on top? I know you like `valgrind` telling you about unset things, but there is no extra information to be had here, other than "oops forgot to set".

Reply:
I'd even go as far and say the caller should set whatever special value he is expecting for unset.

Reply:
I was aiming for minimal change, but I think you're right.  Making the caller set it is a bit too far: this is effectively a trinary return: not found, found, malformed.  My instincts are to try to split the malformed test off into a separate function, but that's awkward here, hence the weirdness.

Reply:
Actually, I turned them into gotos, which is even clearer IMHO.